src/bit_vec.rs

```diff
@@ -50,21 +67,48 @@ impl<O: BitOrder, T: BitStore + ToByteSlice> Encode for BitVec<O, T> {
 			"Attempted to serialize a collection with too many elements.",
 		);
 		Compact(len as u32).encode_to(dest);
-		dest.write(self.as_slice().as_byte_slice());
+
+		let byte_slice: &[u8] = self.as_slice().as_byte_slice();
+
+		if cfg!(target_endian = "big") && mem::size_of::<T>() > 1 {
+			let mut vec_u8: Vec<u8> = byte_slice.into();
+			reverse_endian(&mut vec_u8, mem::size_of::<T>());
+			dest.write(&vec_u8);
+		} else {
+			dest.write(byte_slice);
+		}
 	}
 }
 
 impl<O: BitOrder, T: BitStore + ToByteSlice> EncodeLike for BitVec<O, T> {}
 
+/// # WARNING
+///
+/// In bitvec v0.17.4 the only implementation of BitStore are u8, u16, u32, u64, usize.
+/// This implementation actually only support u8, u16, u32 and u64, encoding of with BitStore being
+/// usize is inconsistent between platforms.
 impl<O: BitOrder, T: BitStore + FromByteSlice> Decode for BitVec<O, T> {
 	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
 		<Compact<u32>>::decode(input).and_then(move |Compact(bits)| {
 			let bits = bits as usize;
 			let required_bytes = required_bytes::<T>(bits);
 
-			let vec = read_vec_u8(input, required_bytes)?;
+			let mut vec_u8 = read_vec_from_u8s::<I, u8>(input, required_bytes)?;
+
```

Comment:
Would moving 'reverse_endian' on aligned vec be better (did we test miri with a big endian target, wondering if reverse_endian can trigger the warning?)?

Reply:
reverse_endian is just changing bytes in a `&mut [u8]` so I think it is fine here.
I havn't run tests in BE actually :-|